verification/testing/deriveContext.go

```diff
@@ -0,0 +1,322 @@
+	if resp, err = c.DeriveContext(handle, make([]byte, digestLen), client.MakeDefault, 0, 0); err != nil {
+		t.Errorf("[ERROR]: Error while creating child context handle: %s", err)
+	}
+	handle = &resp.NewContextHandle
+	if resp.NewContextHandle != client.DefaultContextHandle {
+		t.Fatalf("[FATAL]: Incorrect handle. Should return %v, but returned %v", client.DefaultContextHandle, *handle)
+	}
+
+	// When there is already a default handle, setting MakeDefault and RetainParentContext will cause error
+	// because there cannot be two default handles in a locality
+	if _, err = c.DeriveContext(handle, make([]byte, digestLen), client.MakeDefault|client.RetainParentContext, 0, 0); err == nil {
+		t.Errorf("[ERROR]: Should return %q, but returned no error", client.StatusInvalidArgument)
+	} else if !errors.Is(err, client.StatusInvalidArgument) {
+		t.Errorf("[ERROR]: Incorrect error type. Should return %q, but returned %q", client.StatusInvalidArgument, err)
+	}
+
+	// Retain parent should fail because parent handle is a default handle
+	// and child handle will be a non-default handle.
+	// Default and non-default handle cannot coexist in same locality.
+	if _, err = c.DeriveContext(handle, make([]byte, digestLen), client.RetainParentContext, 0, 0); err == nil {
+		t.Errorf("[ERROR]: Should return %q, but returned no error", client.StatusInvalidArgument)
+	} else if !errors.Is(err, client.StatusInvalidArgument) {
+		t.Errorf("[ERROR]: Incorrect error type. Should return %q, but returned %q", client.StatusInvalidArgument, err)
+	}
+
+	// Child context handle should be a random handle when MakeDefault flag is NOT used
+	if resp, err = c.DeriveContext(handle, make([]byte, digestLen), 0, 0, 0); err != nil {
+		t.Errorf("[ERROR]: Error while creating child context handle: %s", err)
+	}
+	handle = &resp.NewContextHandle
+	if resp.NewContextHandle == client.DefaultContextHandle {
+		t.Fatalf("[FATAL]: Incorrect handle. Should return non-default handle, but returned %v", *handle)
+	}
+	if resp.ParentContextHandle != client.RetiredContextHandle {
+		t.Errorf("[ERROR]: Incorrect handle. Should be invalidated when retain parent is NOT set, but returned %v", resp.ParentContextHandle)
+	}
+
+	// Now, there is no default handle, setting RetainParentContext flag should succeed
+	if resp, err = c.DeriveContext(handle, make([]byte, digestLen), client.RetainParentContext, 0, 0); err != nil {
+		t.Errorf("[ERROR]: Error while making child context handle as default handle: %s", err)
+	}
+	handle = &resp.NewContextHandle
+	parentHandle := resp.ParentContextHandle
+
+	if parentHandle == client.RetiredContextHandle {
+		t.Errorf("[ERROR]: Incorrect handle. Should return retained handle, but returned invalidated handle %v", parentHandle)
+	}
+}
+
+// Validates DerivedChild command with ChangeLocality flag.
+func TestChangeLocality(d client.TestDPEInstance, c client.DPEClient, t *testing.T) {
```

Comment:
I think this test is irreversible as-is. To change this we would need to derive a non-default context that we can destroy before making derive calls.

We can open an issue for that and just mark as irreversible for now.

Reply:
Done, opened https://github.com/chipsalliance/caliptra-dpe/issues/308